core/server/common/src/main/java/alluxio/master/journal/raft/JournalStateMachine.java

```diff
@@ -314,10 +506,11 @@ public boolean isSnapshotting() {
     return mSnapshotting;
   }
 
-  /**
-   * Closes the journal state machine, causing all further modification requests to be ignored.
-   */
-  public void close() {
-    mClosed = true;
+  @Override
+  public void notifyLeaderChanged(RaftGroupMemberId groupMemberId, RaftPeerId raftPeerId) {
+    if (mRaftGroupId == groupMemberId.getGroupId()) {
```

Comment:
What about if this is not true? What would that happen, and do we want to log some sort of warning when that happens?

Reply:
Yeah this should always be true given we only have one raft group. I will add a warning.